lib/domain/plan.js

```diff
@@ -1,9 +1,30 @@
 export default class Plan {
-  constructor({ id, created, firstName, lastName, systemIds }) {
+  constructor({ id, firstName, lastName, systemIds }) {
     this.id = id;
-    this.created = created;
+    this.created = new Date(Date.now()).toISOString();
```

Comment:
This will prevent us from creating `Plan` objects that already have a created date, for example when fetching them from the database. I agree it's nice to have the logic for setting the date inside of the object itself but we also need to account for other scenarios.

Maybe we add a static method for either hydrating from existing plans, or for creating new plans that just calls the constructor with current date?

Reply:
Could we just set created if it's not set already?

Reply:
We could also do that, yeah. I'd be cautious of that approach though in that in the future if the logic for hydrating an existing plan vs. creating a new one becomes more complex we'll end up with this constructor being pretty complex with a bunch of conditional checks etc. when a separate method with a single responsibility would have been easier to maintain.

Reply:
Gotcha